### ISSUE TITLE
p4 not found on PATH

### ISSUE THREAD

**Issue by mil-ad** [mattsawyer77/atom-perforce#65]
Hi,

I have not managed to make atom-perforce find the p4 binary. It's definitely on my PATH and I have specified the path in atom-perforce settings. I use it all the time from terminal.

The console output in Atom gives me this:

```
Error: Command failed: /bin/sh -c p4 info  
/bin/sh: p4: command not found
```

When I run this manually in my terminal it works just fine.

I tried looking at process.env.PATH and it does include the path to p4 binary.

Any suggestions?

**Comment by unional** [comment]
@mattsawyer77 , I'm not familiar with shell. Any input?

**Comment by mattsawyer77** [comment]
/bin/sh is the default shell for unix node.js, regardless of /etc/passwd. But environment.js is supposed to read your shell environment config file and use that when executing commands, even though it's executing them with /bin/sh. What's your shell? And where is p4 on your machine? Do you have the path to it configured in your shell init file?

**Comment by mil-ad** [comment]
@mattsawyer77 I'm using zsh but I tried starting atom from sh and it didn't make a difference. My PATH is defined in my .zshrc

My p4 binary is in a random location but I tried moving it into the default /usr/local/bin but it made no difference.

**Comment by mattsawyer77** [comment]
that's odd. can you try a couple of things:
1. put your export PATH in ~/.zshenv
2. if that didn't work, try launching atom from the terminal

**Comment by mil-ad** [comment]
That made no difference. I have no experience with javascript so I don't know where it's looking for PATH but if it's process.env.PATH then I can see in the debugger that it's the correct one and includes p4 binary. Strange.

**Comment by artyomtrityak** [comment]
@mi-lad open atom from terminal, assume you are in the project root folder 

```
atom .
```

**Comment by flatiron32** [comment]
I am having the same issue using bash. When I execute `/bin/sh -c 'p4 info'` from Terminal it works just fine.

**Comment by mattsawyer77** [comment]
@flatiron32 can you check whether p4 is found on the path as defined in `~/.bash_profile` ?

**Comment by flatiron32** [comment]
I modified my `~/.bash_profile` to echo the path and which p4. 

```
/Users/jtomaw/.rvm/gems/ruby-1.9.3-p194/bin:/Users/jtomaw/.rvm/gems/ruby-1.9.3-p194@global/bin:/Users/jtomaw/.rvm/rubies/ruby-1.9.3-p194/bin:bin:/opt/boxen/homebrew/bin:/opt/boxen/homebrew/sbin:/opt/boxen/bin:bin:/opt/boxen/homebrew/bin:/opt/boxen/homebrew/sbin:/opt/boxen/bin:bin:/opt/boxen/homebrew/bin:/opt/boxen/homebrew/sbin:/opt/boxen/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Users/jtomaw/.rvm/bin:/Users/jtomaw/.rvm/bin:/Users/jtomaw/.rvm/bin
/opt/boxen/homebrew/bin/p4
```

**Comment by hspazio** [comment]
Hope I'm not too late to the party :) 
I've struggled for a day trying to get Atom recognizing the PATH var. I had the PATH exported into `~/.bash_profile` and also the location defined in the settings. Nothing was working so I opened the source code and after some debugging I found out that the PATH variable is not parsed, hence not passed to the command line. 

Here is my fix (file `~/.atom/packages/atom-perforce/lib/atom-perforce.js`): Add 'PATH' to the array.
```
envVarsToExtract = [
        'P4CLIENT',
        'P4CONFIG',
        'P4IGNORE',
        'P4PORT',
        'P4USER',
        'P4TICKETS',
        'P4PASSWD',
        'P4CHARSET',
        'HOME',
        'PATH'
    ],
```

and all commands should work!

@mattsawyer77 was PATH not extracted intentionally or is it a bug?
